error: use of undeclared identifier 'SettingsImpl__getInstallationDirectoryPath'; did you mean 'Settings_getInstallationDirectoryPath'?

Build fails with this error:
```
/disk-samsung/freebsd-ports/misc/openmodelica/work/OpenModelica-1.20.0/OMCompiler/Compiler/runtime/Settings_omc.cpp:46:22: error: use of undeclared identifier 'SettingsImpl__getInstallationDirectoryPath'; did you mean 'Settings_getInstallationDirectoryPath'?
  const char *path = SettingsImpl__getInstallationDirectoryPath();
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                     Settings_getInstallationDirectoryPath
/disk-samsung/freebsd-ports/misc/openmodelica/work/OpenModelica-1.20.0/OMCompiler/Compiler/runtime/Settings_omc.cpp:44:20: note: 'Settings_getInstallationDirectoryPath' declared here
extern const char* Settings_getInstallationDirectoryPath()
                   ^
1 error generated.
```

Replacing the name as suggested by the compiler helps.

--- a/OMCompiler/Compiler/runtime/System_omc.c
+++ b/OMCompiler/Compiler/runtime/System_omc.c
@@ -37,7 +37,7 @@ extern "C"
 {
 #endif
 
-#if defined(__linux__) && !defined(_GNU_SOURCE)
+#if (defined(__linux__) || defined(__FreeBSD__)) && !defined(_GNU_SOURCE)
 #define _GNU_SOURCE 1
 #endif
 

--- a/OMCompiler/Compiler/runtime/settingsimpl.c
+++ b/OMCompiler/Compiler/runtime/settingsimpl.c
@@ -90,7 +90,7 @@ const char* SettingsImpl__getInstallationDirectoryPath(void) {
   return path &&*path ? path : "OPENMODELICA_BOOTSTRAPPING_STAGE_NO_OPENMODELICAHOME";
 }
 #else
-#if (defined(__linux__) || defined(__APPLE_CC__))
+#if (defined(__linux__) || defined(__APPLE_CC__) || defined(__FreeBSD__))
 /* Helper function to strip /bin/... or /lib/... from the executable path of omc */
 static void stripbinpath(char *omhome)
 {
@@ -112,7 +112,7 @@ static void stripbinpath(char *omhome)
 #endif
 
 /* Do not free or modify the returned variable of getInstallationDirectoryPath. It's part of the environment! */
-#if defined(__linux__) || defined(__APPLE_CC__)
+#if defined(__linux__) || defined(__APPLE_CC__)  || defined(__FreeBSD__)
 #include <dlfcn.h>
 
 const char* SettingsImpl__getInstallationDirectoryPath(void) {

--- a/OMCompiler/SimulationRuntime/c/meta/meta_modelica_builtin.c
+++ b/OMCompiler/SimulationRuntime/c/meta/meta_modelica_builtin.c
@@ -705,7 +705,7 @@ modelica_boolean setStackOverflowSignal(modelica_boolean inSignal)
   return inSignal;
 }
 
-#if defined(__linux__) || defined(__APPLE_CC__)
+#if defined(__linux__) || defined(__APPLE_CC__) || defined(__FreeBSD__)
 #include <execinfo.h>
 metamodelica_string referenceDebugString(modelica_metatype fnptr)
 {

--- a/OMCompiler/SimulationRuntime/c/meta/meta_modelica_segv.c
+++ b/OMCompiler/SimulationRuntime/c/meta/meta_modelica_segv.c
@@ -30,7 +30,7 @@
 
 /* Stack overflow handling */
 
-#if defined(__linux__) && !defined(_GNU_SOURCE)
+#if (defined(__linux__) || defined(__FreeBSD__)) && !defined(_GNU_SOURCE)
 #define _GNU_SOURCE 1
 /* for pthread_getattr_np */
 #endif
@@ -57,7 +57,7 @@ int mmc_hasStacktraceMessages(threadData_t *threadData)
 
 pthread_key_t mmc_stack_overflow_jumper;
 
-#if defined(__linux__) || defined(__APPLE_CC__)
+#if defined(__linux__) || defined(__APPLE_CC__) || defined(__FreeBSD__)
 #include <stdlib.h>
 #include <stdio.h>
 #include <signal.h>

--- a/OMCompiler/SimulationRuntime/c/simulation/solver/real_time_sync.c
+++ b/OMCompiler/SimulationRuntime/c/simulation/solver/real_time_sync.c
@@ -30,7 +30,7 @@
 
 #include "real_time_sync.h"
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__FreeBSD__)
 #include <sys/mman.h>
 #include <errno.h>
 #endif
@@ -45,7 +45,7 @@ void omc_real_time_sync_init(threadData_t *threadData, DATA *data)
     return;
   }
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__FreeBSD__)
   if (mlockall(MCL_CURRENT | MCL_FUTURE) == -1) {
     warningStreamPrint(LOG_RT, 0, __FILE__ ": mlockall failed (recommended to run as root to lock memory into RAM while doing real-time simulation): %s\n", strerror(errno));
   }

--- a/OMCompiler/SimulationRuntime/c/util/rtclock.c
+++ b/OMCompiler/SimulationRuntime/c/util/rtclock.c
@@ -431,7 +431,7 @@ int64_t rt_ext_tp_sync_nanosec(rtclock_t* tick_tp, uint64_t nsec)
 static clockid_t omc_clock = OMC_CLOCK_MONOTONIC;
 
 int rt_set_clock(enum omc_rt_clock_t newClock) {
-#if defined(__linux__)
+#if defined(__linux__) || defined(__FreeBSD__)
   omc_clock = newClock == OMC_CLOCK_REALTIME ? OMC_CLOCK_MONOTONIC : CLOCK_PROCESS_CPUTIME_ID;
 #else
   omc_clock = OMC_CLOCK_MONOTONIC;

